src/main/features/boot.json

```diff
@@ -145,16 +145,15 @@
+        // export XML packages in addition to JRE version with special versions for backwards-compatibility reasons
+        "sling.jre-1.8.javax.xml":",javax.xml;version=\"2.1.0\",javax.xml.datatype;uses:=\"javax.xml.namespace\";version=\"2.1.0\",javax.xml.namespace;version=\"2.1.0\",javax.xml.parsers;uses:=\"javax.xml.validation,org.w3c.dom,org.xml.sax,org.xml.sax.helpers\";version=\"2.1.0\",javax.xml.stream;uses:=\"javax.xml.namespace,javax.xml.stream.events,javax.xml.stream.util,javax.xml.transform\";version=\"1.0.0\",javax.xml.stream.events;uses:=\"javax.xml.namespace,javax.xml.stream\";version=\"1.0.0\",javax.xml.stream.util;uses:=\"javax.xml.namespace,javax.xml.stream,javax.xml.stream.events\";version=\"1.0.0\",javax.xml.transform;version=\"2.1.0\",javax.xml.transform.dom;uses:=\"javax.xml.transform,org.w3c.dom\";version=\"2.1.0\",javax.xml.transform.sax;uses:=\"javax.xml.transform,javax.xml.transform.stream,org.xml.sax,org.xml.sax.ext\";version=\"2.1.0\",javax.xml.transform.stax;uses:=\"javax.xml.stream,javax.xml.stream.events,javax.xml.transform\";version=\"2.1.0\",javax.xml.transform.stream;uses:=\"javax.xml.transform\";version=\"2.1.0\",javax.xml.validation;uses:=\"javax.xml.transform,javax.xml.transform.stream,org.w3c.dom,org.w3c.dom.ls,org.xml.sax\";version=\"2.1.0\",javax.xml.xpath;uses:=\"javax.xml.namespace,org.xml.sax\";version=\"2.1.0\",org.w3c.dom;version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.bootstrap;uses:=\"org.w3c.dom\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.css;uses:=\"org.w3c.dom,org.w3c.dom.stylesheets,org.w3c.dom.views\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.events;uses:=\"org.w3c.dom,org.w3c.dom.views\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.html;uses:=\"org.w3c.dom\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.ls;uses:=\"org.w3c.dom,org.w3c.dom.events\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.ranges;uses:=\"org.w3c.dom\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.stylesheets;uses:=\"org.w3c.dom\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.traversal;uses:=\"org.w3c.dom\";version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.views;version=\"{dollar}{felix.detect.java.version}\",org.w3c.dom.xpath;uses:=\"org.w3c.dom\";version=\"{dollar}{felix.detect.java.version}\",org.xml.sax;version=\"{dollar}{felix.detect.java.version}\",org.xml.sax.ext;uses:=\"org.xml.sax,org.xml.sax.helpers\";version=\"{dollar}{felix.detect.java.version}\",org.xml.sax.helpers;uses:=\"org.xml.sax\";version=\"{dollar}{felix.detect.java.version}\"",
```

Comment:
@karlpauls Can I remove all uses directives? Is Felix calculating those automatically?

Reply:
@kwin, it only calculates them for jpms - so for java <= 8 you have to add them :-(